pkgs/by-name/so/soundconverter/package.nix

```
@@ -0,0 +1,117 @@
+{
+  lib,
+  fetchurl,
+  # Optional due to unfree license.
+  faacSupport ? false,
+  glib,
+  python3Packages,
+  gtk3,
+  wrapGAppsHook,
+  python3,
+  gsettings-desktop-schemas,
+  intltool,
+  xvfb-run,
+  gobject-introspection,
+  gst_all_1,
+  fdk-aac-encoder,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "soundconverter";
+  version = "4.0.5";
+
+  src = fetchurl {
+    url = "https://launchpad.net/soundconverter/trunk/${version}/+download/${pname}-${version}.tar.gz";
+    sha256 = "1wvrxcvbhgqx9kliz5v2kvc4w08aw043v8ph4kgfbp2xd0v5mc3r";
+  };
+
+  buildInputs = [
+    gtk3
+    fdk-aac-encoder
+    gst_all_1.gst-libav
+    gst_all_1.gst-plugins-base
+    gst_all_1.gst-plugins-good
+    gst_all_1.gst-plugins-ugly
+    (gst_all_1.gst-plugins-bad.override { inherit faacSupport; })
+  ];
+
+  nativeBuildInputs = [
+    intltool
+    wrapGAppsHook
+    gobject-introspection
+  ];
+
+  propagatedBuildInputs = [
+    python3Packages.gst-python
+    python3Packages.distutils-extra
+    python3Packages.setuptools
+    python3Packages.pygobject3
+  ];
+
+  nativeCheckInputs = [ xvfb-run ];
+
+  postPatch = ''
+    substituteInPlace  bin/soundconverter --replace \
+      "DATA_PATH = os.path.join(SOURCE_PATH, 'data')" \
+      "DATA_PATH = '$out/share/soundconverter'"
+  '';
+
+  preCheck =
+    let
+      self = {
+        outPath = "$out";
+        name = "${pname}-${version}";
+      };
+      xdgPaths = lib.concatMapStringsSep ":" glib.getSchemaDataDirPath;
+    in
+    ''
+      export HOME=$TMPDIR
+      export XDG_DATA_DIRS=$XDG_DATA_DIRS:${
+        xdgPaths [
+          gtk3
+          gsettings-desktop-schemas
+          self
+        ]
+      }
+      # FIXME: Fails due to weird Gio.file_parse_name() behavior.
+      sed -i '49 a\    @unittest.skip("Gio.file_parse_name issues")' tests/testcases/names.py
+    ''
+    + lib.optionalString (!faacSupport) ''
+      substituteInPlace tests/testcases/gui_integration.py --replace \
+        "for encoder in ['fdkaacenc', 'faac', 'avenc_aac']:" \
+        "for encoder in ['fdkaacenc', 'avenc_aac']:"
+    '';
+
+  checkPhase = ''
+    runHook preCheck
+    xvfb-run python tests/test.py
+    runHook postCheck
+  '';
+
+  # Fixes https://ryantm.github.io/nixpkgs/languages-frameworks/gnome/#ssec-gnome-common-issues-double-wrapped
+  dontWrapGApps = true;
+
+  preFixup = ''
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+    makeWrapperArgs+=(--prefix PYTHONPATH : ${python3.pkgs.makePythonPath propagatedBuildInputs})
```

```suggestion
    makeWrapperArgs+=(--prefix PYTHONPATH : ${python3.pkgs.makePythonPath propagatedBuildInputs})
```

why was this line needed?

---

This fixes an issue with PyGobject not initializing properly.